docs/framework/architecture/ui-library.md

```diff
@@ -439,6 +441,159 @@ dropdownView.bind( 'isEnabled' ).toMany( buttons, 'isEnabled',
+
+A dialog is a popup window that does not close when the user clicks outside it, so it allows to interact with the editor and its content while being open (unless it is a modal, which blocks the interaction with the rest of the page until it is closed). As long as it has a [header](#header), it is also draggable. Only one dialog can be open at a time - opening another one closes the previously visible window.
+
+A minimal plugin creating a toolbar button component showing an empty dialog can look like this:
+
+```js
+import { Dialog, ButtonView } from '@ckeditor/ckeditor5-ui';
+
+class MyPluginWithDialog extends Plugin {
+	public static get requires() {
+		return [ Dialog ] as const;
+	}
+
+	public init(): void {
+		this.editor.ui.componentFactory.add( 'aiAssistant', locale => {
+			const button = new ButtonView( locale );
+
+			button.set( {
+				label: 'Show a dialog'
+			} );
+
+			this.listenTo( button, 'execute', () => {
+				const dialog = this.editor.plugins.get( 'Dialog' );
+
+				dialog.show( {} );
+			} );
+
+			return button;
+		} );
+	}
+}
+```
+
+Modals are very similar to the dialogs - they share the same structure rules, API etc. The major difference is that while a modal is open, the user can not interact with the editor or the rest of the page - it is covered with a non-transparent overlay. It can be used e.g. to enforce user to take one of the specified actions.
+
+To create a modal, use the optional {@link module:ui/dialog/dialog~DialogDefinition#isModal `isModal`} argument of the {@link module:ui/dialog/dialog~Dialog#show `Dialog#show()`} method:
+
+```js
+dialog.show( {
+	isModal: true
+} );
+```
+
+Always make sure there is a way for the user to close a modal - either via a "Close" button, <kbd>Esc</kbd> key or one of the action buttons.
+
+#### Structure and behaviour
+
+A {@link module:ui/dialog/dialogview~DialogView `DialogView`} can have three parts, each of which is optional: the header (which is used as a drag handler), the contents (the body of the dialog) and the actions area (a collection of buttons). The order of the parts can not be changed.
+
+##### Header
+
+A header may consist of any combination of three elements:
+
+* an icon;
+* a title;
+* a "Close" button.
+
+The "Close" button is added automatically as long as you provide an icon or a title. The code below shows how to create a header with an icon and a title, but without a "Close" button.
+
+```js
+dialog.show( {
+	icon: pencilIcon,
+	title: 'My first dialog',
+	hasCloseButton: false
+} );
+```
+
+##### Contents
+
+This part can be a single [view](#views) or a collection of views. They will be displayed directly inside a body of a dialog.
```

Comment:
What I'm missing here is an example of how to inject any view into the dialog. The example in the first section is not helpful because it does not bring any content. I'd either extend the initial example or create another one with some content (e.g. simple paragraph view with some text) here.

The whole point of docs is to give developers a minimal snippet to achieve something at least somehow relateable to their goals. One thing I'm sure they'll want to display _something_ in their dialog.

Reply:
+1. At least add some simple view with some text, like in manual tests.

Reply:
I'd also add an information, here or somewhere, that the dialog size will depend on the view passed to it. Dialog itself does not have any settings related to size, so it is up to the integrator to create a view with a proper size.

Also, if the internal view size has changed it may be necessary to call `updatePositon()` -- I'd mention that. Unless we make this automatic. Which would be great.